client/js/editmode.js

```diff
@@ -871,13 +871,17 @@ async function updateWidget(currentState, oldState, applyChangesFromUI) {
         widget[key] = null;
   }
 
-  const id = await addWidgetLocal(widget);
+  if(widget.id !== previousState.id || widget.type !== previousState.type) {
+    const id = await addWidgetLocal(widget);
 
-  if(widget.id !== previousState.id) {
     for(const child of children)
       sendPropertyUpdate(child.get('id'), 'parent', id);
     for(const card of cards)
       sendPropertyUpdate(card.get('id'), 'deck', id);
+  } else {
+    for(const key in widget)
+      if(widget[key] !== previousState[key])
+        sendPropertyUpdate(widget.id, key, widget[key]);
```

Comment:
This seems to add functionality that wasn't there before. Is this more than a performance change?

Reply:
This is the long-awaited change that editing a widget doesn't trigger an update to _all_ its properties but only the _changed_ properties.

It doesn't work with objects yet. I gotta add a second check with `JSON.stringify`.

Reply:
I did a few more tests and refined it a bit to further decrease false positives.

I hope this doesn't break in any case but either way it's only used by the editors.

Reply:
It *looks* right.